pymongo/srv_resolver.py

```diff
@@ -97,6 +97,9 @@ def _resolve_uri(self, encapsulate_errors):
 
     def _get_srv_response_and_hosts(self, encapsulate_errors):
         results = self._resolve_uri(encapsulate_errors)
+        if (self.__srv_max_hosts != 0 and self.__srv_max_hosts is not None) \
+                and self.__srv_max_hosts <= len(results):
+            results = sample(results, self.__srv_max_hosts)
```

Comment:
Is this what the spec says to do? The way this PR is currently written, the driver will completely replace all the hosts with a new random set of hosts every time the SRV record gets polled (every 60 seconds I think). That does not sound like a good idea to me.

Reply:
I'm not sure how to find the relevant section of the spec. I took a look around in Jira but could not find any reference to a specification for this behavior.